mpc-core/src/protocols/rep3/gadgets/sort.rs

```diff
@@ -0,0 +1,34 @@
+//! Sort
+//!
+//! This module contains some oblivious sorting algorithms for the Rep3 protocol.
+
+use crate::protocols::rep3::{
+    arithmetic::FieldShare,
+    network::{IoContext, Rep3Network},
+    yao::{self, circuits::GarbledCircuits},
+    IoResult,
+};
+use ark_ff::PrimeField;
+
+/// Sorts the inputs using the Batcher's odd-even merge sort algorithm. Thereby, only the lowest `bitsize` bits are considered. The final results also only hav bitsize bits each.
+pub fn batcher_odd_even_merge_sort_yao<F: PrimeField, N: Rep3Network>(
```

Comment:
yes the field elements are also pushed into the trace, so we need the decomposed elements as field elements. We can maybe later optimize this away that the intermediate result is also stored in bits, but this will not be easy...